android-exoplayer/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java

```diff
@@ -300,6 +361,84 @@ private MediaSource buildMediaSource(Uri uri, String overrideExtension) {
         }
     }
 
+    private DefaultDrmSessionManager createDrmSessionManager(ActionToken drmParam) throws UnsupportedDrmException {
+
+        if (drmParam == null) {
+            return null;
+        }
+
+        DefaultDrmSessionManager<FrameworkMediaCrypto> drmSessionManager = null;
+        final String drmLicenseUrl = drmParam.getLicensingServerUrl();
+        final HashMap<String, String> keyRequestProperties = Utils.getParams(drmParam);
+
+        UnsupportedDrmException exception = null;
+        if (Util.SDK_INT < 18) {
+            throw new UnsupportedDrmException(0);
+        } else {
+            UUID drmSchemeUuid = Util.getDrmUuid(drmParam.getDrmScheme());
+            if (drmSchemeUuid == null) {
+                throw new UnsupportedDrmException(UnsupportedDrmException.REASON_UNSUPPORTED_SCHEME);
+            } else {
+                drmSessionManager = buildDrmSessionManagerV18(drmSchemeUuid, drmLicenseUrl, keyRequestProperties, false);
+
+                final String offline = drmParam.getOfflineLicense();
+
+                if (offline != null) {
+                    drmSessionManager.setMode(DefaultDrmSessionManager.MODE_QUERY, Base64.decode(offline, Base64.NO_WRAP));
+                }
+            }
+        }
+        return drmSessionManager;
+    }
+
+    private void handleDrmError(UnsupportedDrmException exception) {
+        final int errorStringId;
+        switch (exception.reason) {
+            case 0:
+                errorStringId = com.dice.shield.downloads.R.string.error_drm_not_supported;
+                break;
+            case UnsupportedDrmException.REASON_UNSUPPORTED_SCHEME:
+                errorStringId = com.dice.shield.downloads.R.string.error_drm_unsupported_scheme;
+                break;
+            default:
+                errorStringId = com.dice.shield.downloads.R.string.error_drm_unknown;
+                break;
+        }
+
+        final String errorString = getContext().getString(errorStringId);
+        eventEmitter.error(errorString, exception);
+    }
+
+    private DefaultDrmSessionManager<FrameworkMediaCrypto> buildDrmSessionManagerV18(
+            UUID uuid, String licenseUrl, HashMap<String, String> requestProperties, boolean multiSession)
```

Comment:
Could it be a `Map<String, String>` so it's not a specifc implemntation of Maps?

Reply:
Done.